torch/csrc/autograd/variable.cpp

```diff
@@ -119,34 +120,34 @@ std::shared_ptr<Function> Variable::Impl::get_grad_accumulator() {
   if (result)
     return result;
 
-  c10::raw::intrusive_ptr::incref(this);
-  auto intrusive_from_this = c10::intrusive_ptr<Variable::Impl>::reclaim(this);
+  c10::raw::intrusive_ptr::incref(unsafeGetTensorImpl());
+  auto intrusive_from_this = c10::intrusive_ptr<Variable::Impl>::reclaim(static_cast<Variable::Impl*>(unsafeGetTensorImpl()));
   result = std::make_shared<AccumulateGrad>(Variable(std::move(intrusive_from_this)));
   autograd_meta->grad_accumulator_ = result;
   return result;
 }
 
-void Variable::Impl::detach_() {
-  auto autograd_meta = get_autograd_meta();
-  if (autograd_meta->is_view_) {
+void Variable::detach_() {
+  if (is_view()) {
     AT_ERROR("Can't detach views in-place. Use detach() instead");
   }
-  set_requires_grad(false);
+  auto autograd_meta = get_autograd_meta();
+  autograd_meta->set_requires_grad(false, unsafeGetTensorImpl());
   autograd_meta->grad_fn_.reset();
   autograd_meta->output_nr_ = 0;
 }
 
-void Variable::Impl::backward(
+void Variable::backward(
     c10::optional<Tensor> gradient,
     bool keep_graph,
-    bool create_graph) {
+    bool create_graph) const {
   auto autograd_meta = get_autograd_meta();
   std::vector<Edge> edges;
   edges.emplace_back(autograd_meta->grad_fn_, autograd_meta->output_nr_);
 
   std::vector<Variable> inputs;
   if (!gradient.has_value()) {
-    gradient = make_variable(at::ones_like(data_), /*requires_grad=*/false);
+    gradient = make_variable(at::ones_like(data()), /*requires_grad=*/false);
```

Comment:
is this future proofing or is there a semantic change here?

Reply:
There is actually no semantic change - the old `backward` function lives in Variable::Impl and we can directly access `data_`, while the new `backward` function lives in Variable and we need to use either `data()` or `impl_->data_` to access the underlying Tensor data. 

In the next PR (when Variable itself becomes the data tensor), we will use `*this` instead of `data()` in this line.